plugin/cog/cog-commons/src/main/java/it/geosolutions/imageioimpl/plugins/cog/AbstractRangeReader.java

```diff
@@ -68,11 +69,14 @@ public AbstractRangeReader(BasicAuthURI authUri, int headerLength) {
                     long[] newRange = new long[]{headerLength - 1, ranges[i][1]};
 
                     // Adjust the data's header size in case the related range have been adjusted
-                    byte[] headersData = data.get(0L);
-                    if (headersData != null && headerLength < headersData.length) {
-                        byte[] newHeader = new byte[headerLength];
-                        System.arraycopy(headersData, 0, newHeader, 0, headerLength);
-                        data.put(0L, newHeader);
+
+                    synchronized (data) {
```

Comment:
SoftValueHashMap is thread safe, is the synch needed?

Reply:
Nope. that's the only one I have forgot to remove.